File: app/src/main/java/com/arjanvlek/oxygenupdater/download/DownloadStatus.java
package com.arjanvlek.oxygenupdater.download;

/**
 * Oxygen Updater, copyright 2019 Arjan Vlek. File created by arjan.vlek on 05/01/2019.
 */
public enum DownloadStatus {

    /**
     * No update is being downloaded and the update has not been downloaded yet.
     */
    NOT_DOWNLOADING,

    /**
     * The download is in the queue of the download executor, it will start soon.
     */
    DOWNLOAD_QUEUED,

    /**
     * The download is in progress
     */
    DOWNLOADING,

    /**
     * The download has been paused by the user
     */
    DOWNLOAD_PAUSED,

    /**
     * The download has been paused due to the loss of network connectivity
     */
    DOWNLOAD_PAUSED_WAITING_FOR_CONNECTION,

    /**
     * The downloaded file is being verified (MD5 check)
     */
    VERIFYING,

    /**
     * The file has been successfully downloaded
     */
    DOWNLOAD_COMPLETED
}

File: app/src/main/java/com/arjanvlek/oxygenupdater/enums/DownloadStatus.kt
package com.arjanvlek.oxygenupdater.enums

/**
 * @author [Adhiraj Singh Chauhan](https://github.com/adhirajsinghchauhan)
 * @author [Arjan Vlek](https://github.com/arjanvlek)
 */
enum class DownloadStatus {
    /**
     * No update is being downloaded and the update has not been downloaded yet.
     */
    NOT_DOWNLOADING,

    /**
     * The download is in the queue of the download executor, it will start soon.
     */
    DOWNLOAD_QUEUED,

    /**
     * The download is in progress
     */
    DOWNLOADING,

    /**
     * The download has been paused by the user
     */
    DOWNLOAD_PAUSED,

    /**
     * The file has been successfully downloaded
     */
    DOWNLOAD_COMPLETED,

    /**
     * The file could not be downloaded
     */
    DOWNLOAD_FAILED,

    /**
     * The downloaded file's MD5 checksum is being verified
     */
    VERIFYING,

    /**
     * The downloaded file's MD5 checksum has been verified
     */
    VERIFICATION_COMPLETED,

    /**
     * The downloaded file's MD5 checksum could not be verified
     */
    VERIFICATION_FAILED;

    fun shouldPruneWork() = this == DOWNLOAD_COMPLETED || this == DOWNLOAD_FAILED
            || this == VERIFICATION_COMPLETED || this == VERIFICATION_FAILED
}
